### DISCUSSION TITLE
Google signin initialize set up

### DISCUSSION THREAD

**Discussion by Brunopaun** [nuxt/nuxt#27634] [Questions] [answered]
Hello Community,

I've started this discussion to learn about best practices for Vue applications.

Currently, I'm implementing a Google sign-in button. I've noticed that many developers hard-code their Google login client ID, which seems problematic to me from a security standpoint.

I encountered an issue accessing my .env file within the Vue setup context, so I implemented it as shown below. I'd appreciate any feedback or suggestions for alternative approaches.

Additionally, is there a security concern with using config.public for the Google login client ID?

<script setup>
 const config = useRuntimeConfig()
    onMounted(() => {
      google.accounts.id.initialize({
        client_id: `${config.public.GOOGLE_LOGIN_CLIENT}`,
        use_fedcm_for_prompt: true,
        callback: handleGoogleLogin
      })

      const element = document.getElementById('btn')
      google.accounts.id.renderButton(element, {
        theme: 'outline',
        shape: 'pill',
        size: 'medium',
        text: 'continue_with'
      })
    })
 </script>

**Comment by vinhegewald** [comment]
I am not entirely sure but always review your built files (dist directory) to check if sensitive data appears in your JavaScript bundles or HTML. Ideally, sensitive data should not be visible in the final output.

**Comment by meirroth** [comment, marked as answer]
@Brunopaun The Google Client ID can be safely exposed to the client. In fact, it is necessary to include it on the client side to set up the ‘Sign in with Google’ button.

**Reply by Brunopaun** [reply]
Thanks a lot
